debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerSnapshotChangeEventSource.java

```diff
@@ -79,31 +78,31 @@ protected SnapshotContext prepare(ChangeEventSourceContext context) throws Excep
     @Override
     protected void lockTablesForSchemaSnapshot(ChangeEventSourceContext sourceContext, SnapshotContext snapshotContext) throws SQLException, InterruptedException {
         if (connectorConfig.getSnapshotLockingMode() == SnapshotLockingMode.NONE) {
-            jdbcConnection.connection().setTransactionIsolation(Connection.TRANSACTION_REPEATABLE_READ);
-            ((SqlServerSnapshotContext)snapshotContext).preSchemaSnapshotSavepoint = jdbcConnection.connection().setSavepoint("dbz_schema_snapshot");
+            // No table locks acquired. Read committed isolation level is used to avoid taking shared locks on rows.
+            jdbcConnection.connection().setTransactionIsolation(Connection.TRANSACTION_READ_COMMITTED);
             LOGGER.info("Schema locking was disabled in connector configuration");
-            return;
-        }
-        LOGGER.info("Executing schema locking");
+        } else if (connectorConfig.getSnapshotLockingMode() == SnapshotLockingMode.EXCLUSIVE) {
+            LOGGER.info("Executing schema locking");
```

Comment:
Taking locks here is misleading, since it suggests they are taken only for schema snapshot. I believe we should introduce yet another template method.

Reply:
Yes, that's a good point. Can you file a JIRA issue so we can track it separately?

Reply:
Hum, no, I think I'm taking that back. Before this PR, these locks should indeed only have been held while taking the schema snapshot, by virtue of a savepoint. Which makes me wonder why you removed it here?

Reply:
Rollback to savepoint makes no sense.
In both SNAPSHOT and R_R mode no locks are taken - no rollback needed.
For EXCLUSIVE, we want to keep locks until the end of transaction - rollback must not be executed.

Reply:
I meant that taking exclusive lock in `lockTablesForSchemaSnapshot` is misleading.

Reply:
> For EXCLUSIVE, we want to keep locks until the end of transaction - rollback must not be executed.

I _think_ the original idea was to keep the lock only for the time of taking the schema snapshot. At least that was my motivation for the Oracle connector, which is where @jpechane copied this from. Then the actual data snapshot would be taken using R_R semantics, which indeed seems to be missing still, when looking at the master branch. @jpechane Correct me please, if I'm wrong.

Reply:
Yes, on master in EXCLUSIVE data snapshot is taken in database default isolation level.

I suggested to keep the locks during data snapshot as well due to this scenario: https://issues.jboss.org/browse/DBZ-947?focusedCommentId=13651038&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-13651038